### PR TITLE
Business central import settings UI

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new import location setting to enhance import configurations.
  - Added a new type for accounting application default fields to improve data handling.
  - Expanded onboarding steps with a new `LOCATION` state for better user guidance.

- **Enhancements**
  - Improved UI components for import settings, providing a more intuitive user experience.
  - Added new conditional rendering logic to optimize interface responsiveness.
  - Enhanced dropdown styling for better visual clarity in configuration fields.

- **Bug Fixes**
  - Adjusted icon rendering in onboarding steps to ensure correct display of active and inactive states.

- **Style Updates**
  - Updated SVG icons to align with new design specifications for a refreshed look and feel.

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
Screenshot 2023-12-12 at 6 45 22 PM

File: src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.html [hunk 1]
```diff
@@ -1 +1,66 @@
...
+        [customFieldForm]="customFieldForm"
+        [showCustomFieldCreationDialog]="showCustomFieldDialog" 
+        (saveCustomFieldFn)="saveFyleExpenseField()"
+        (closeModelFn)="closeModel()">
+    </app-configuration-custom-field-creation-dialog>
+</div>
+<div *ngIf="isPreviewDialogVisible">
+    <app-preview-dialog 
+        [isPreviewDialogVisible]="true"
+        [iconPath]="'assets/illustrations/sageIntacct/IntacctImportSettings.png'" (closeDialog)="closeDialog()"
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.html:63]: can you rename this image to a generic folder and generic name? confused when I read it

File: src/app/core/models/business-central/business-central-configuration/business-central-import-settings.model.ts [hunk 2]
```diff
@@ -22,7 +22,8 @@ export class BusinessCentralImportSettingsModel {
         const expenseFieldsArray = importSettings?.mapping_settings ? ImportSettingsModel.constructFormArray(importSettings.mapping_settings, businessCentralFields) : [] ;
         return new FormGroup({
             importCategories: new FormControl(importSettings?.import_categories ?? false),
-            expenseFields: new FormArray(expenseFieldsArray)
+            expenseFields: new FormArray(expenseFieldsArray),
+            importLocation: new FormControl(expenseFieldsArray.length > 0 ? true : false)
```
Comment (ashwin1111) [inline src/app/core/models/business-central/business-central-configuration/business-central-import-settings.model.ts:26]: handle UI completely generic similar to other apps, it should handle other field additions in future as well.

File: src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.html [hunk 3]
```diff
@@ -1 +1,66 @@
...
+                        [iconPath]="'arrow-tail-up-down'">
+                    </app-configuration-toggle-field>
+                </div>
+                
+                <div class="tw-rounded-lg tw-border-separator tw-border tw-mt-24-px tw-pb-24-px">
+                    <app-configuration-toggle-field 
+                        [form]="importSettingForm"
+                        [label]="'Import Location from Dynamics 365 Business Central'" 
+                        [subLabel]="'The imported Location from Dynamics 365 Business Central will be a selectable field while creating an expense.' " 
+                        [formControllerName]="'importLocation'"
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.html:31]: handle UI completely generic similar to other apps, it should handle other field additions in future as well.

File: src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.html [hunk 4]
```diff
@@ -1 +1,66 @@
...
+                        [isSectionHeader]="false"
+                        [iconPath]="'arrow_tail'">
+                    </app-configuration-toggle-field>
+                    <div *ngIf="importSettingForm.value.importLocation" class="tw-pl-14-px">
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.html:35]: handle UI completely generic similar to other apps, it should handle other field additions in future as well.

File: src/app/core/models/common/import-settings.model.ts [hunk 5]
```diff
@@ -4,6 +4,12 @@ import { BusinessCentralImportSettingsGet } from "../business-central/business-c
 import { IntegrationField } from "../db/mapping.model";
 import { Sage300ImportSettingGet } from "../sage300/sage300-configuration/sage300-import-settings.model";
 
+export type AccountingAppDefaultFields = {
```
Comment (ashwin1111) [inline src/app/core/models/common/import-settings.model.ts:7]: singular and better name please, call it ImportDefaultField

File: src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.html [hunk 6]
```diff
@@ -1 +1,66 @@
...
+                        [isSectionHeader]="false"
+                        [iconPath]="'arrow-tail-up-down'">
+                    </app-configuration-toggle-field>
+                </div>
+                
+                <div class="tw-rounded-lg tw-border-separator tw-border tw-mt-24-px tw-pb-24-px">
+                    <app-configuration-toggle-field 
+                        [form]="importSettingForm"
+                        [label]="'Import Location from Dynamics 365 Business Central'" 
+                        [subLabel]="'The imported Location from Dynamics 365 Business Central will be a selectable field while creating an expense.' "
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.html:30]: location should be dynamic

File: src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.ts [hunk 7]
```diff
@@ -47,6 +49,16 @@ export class BusinessCentralImportSettingsComponent implements OnInit {
 
   customFieldOption: any;
 
+  readonly brandingConfig = brandingConfig;
+
+  supportArticleLink: string = brandingKbArticles.onboardingArticles.SAGE300.IMPORT_SETTING;
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.ts:54]: readonly

File: src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.ts [hunk 8]
```diff
@@ -47,6 +49,16 @@ export class BusinessCentralImportSettingsComponent implements OnInit {
...
+
+  appName: string = AppName.BUSINESS_CENTRAL;
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-import-settings/business-central-import-settings.component.ts:56]: readonly

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html [hunk 9]
```diff
@@ -1,5 +1,5 @@
 <div [formGroup]="form">
-    <div class="fields-header" fxLayout="row" fxLayoutAlign="start center">
+    <div *ngIf="!isSingleDefaultImportField" class="fields-header" fxLayout="row" fxLayoutAlign="start center">
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html:2]: `isDestinationFixedImport`

**Comment by DhaaraniCIT** [comment]
the comments are fixed in this PR: https://github.com/fylein/fyle-integrations-app/pull/365

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**